### PR TITLE
DALA-1089 Remove custom API-Key Header

### PR DESCRIPTION
**Pull request by MarcTM01**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one E2E Test exists testing the new feature
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] The new version is deployed to the dev server "dev1" using this branch
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] The new feature is manually used/tested/observed on dev server
  - [x] All implemented Social Logins have been tested manually in the UI
- [x] If any work on the UI is to be merged, those changes were also documented in the Figma

### CODE REVIEW

File: dataland-keycloak-adapter/src/main/kotlin/org/dataland/keycloakAdapter/auth/DatalandRealmRole.kt [hunk 1]
```diff
@@ -5,6 +5,6 @@ package org.dataland.keycloakAdapter.auth
  * Datalandsecurity realm. Can be used in conjunction with the
  * Authentication.hasRole extension function to check if a user has a certain role
  */
-enum class DatalandRealmRoles {
+enum class DatalandRealmRole {
```
Comment (fsoffel) [inline dataland-keycloak-adapter/src/main/kotlin/org/dataland/keycloakAdapter/auth/DatalandRealmRole.kt:8]: Does the renaming have a specific reason? I personally prefer the plural version, probably just because I'm used to it since I can not come up with a different reason. The question "What does the type contain?" would be better answered by the plural version but the question "What is it?" is better answered with the singular version. However, in the code we seem to do both. So do what you want. Sorry for the monologue :)

File: dataland-frontend/src/components/pages/ApiKeysPage.vue [hunk 2]
```diff
@@ -85,9 +85,8 @@
             <div>
               <div class="text-900 font-medium text-xl text-left">API Key usage info</div>
               <div class="block text-600 mb-2 mt-4 text-left">
-                In order to use the API Key, include it into the http request header via "dataland-api-key:
-                &lt;ApiKeyValue&gt;". Alternatively it can be used like a bearer token by adding "Authorization: Bearer
-                &lt;ApiKeyValue&gt;" to the http request header.
+                You can use the API-Keys as bearer-tokens by including them in the authorization header of any requests
+                you make to dataland (i.e., "Authorization: Bearer &lt;ApiKeyValue&gt;").
```
Comment (fsoffel) [inline dataland-frontend/src/components/pages/ApiKeysPage.vue:89]: The value seems unneccessary, maybe confusing. I'd prefer reading "Bearer <Api Key>" or the same in lowercase

File: dataland-keycloak-adapter/src/main/kotlin/org/dataland/keycloakAdapter/auth/DatalandApiKeyAuthentication.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ */
+class DatalandApiKeyAuthentication(
+    private val apiKey: String,
+    val apiKeyMetaInformation: ApiKeyMetaInfo,
+) : DatalandAuthentication() {
+    override val userId: String
+        get() = apiKeyMetaInformation.keycloakUserId!!
+
+    override fun getAuthorities(): List<GrantedAuthority> {
+        return apiKeyMetaInformation.keycloakRoles?.map { SimpleGrantedAuthority(it) } ?: emptyList()
```
Comment (fsoffel) [inline dataland-keycloak-adapter/src/main/kotlin/org/dataland/keycloakAdapter/auth/DatalandApiKeyAuthentication.kt:18]: I see that we store these roles alongside with the api key, but can you explain to me why?

File: dataland-keycloak-adapter/src/main/kotlin/org/dataland/keycloakAdapter/config/WebSecurityConfig.kt [hunk 4]
```diff
@@ -72,8 +80,9 @@ class WebSecurityConfig(
             .requestMatchers(*publicLinkMatchers).permitAll()
             .anyRequest().fullyAuthenticated()
             .and()
+            .logout().disable()
```
Comment (fsoffel) [inline dataland-keycloak-adapter/src/main/kotlin/org/dataland/keycloakAdapter/config/WebSecurityConfig.kt:83]: Why that?

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/DataRetrievalViaApiKeyTest.kt [hunk 5]
```diff
@@ -125,51 +126,54 @@ class DataRetrievalViaApiKeyTest {
...
-        apiKeyHandler.obtainApiKeyForUserTypeAndRevokeBearerTokens(userType, 1)
+        val apiKey = apiKeyHelper.obtainApikeyForTechnicalUser(technicalUser, 1)
+        GlobalAuth.setBearerToken(apiKey.apiKey)
         val downloadedStoredCompany = apiAccessor.companyDataControllerApi.getCompanyById(companyId)
         assertEquals(expectedStoredCompany, downloadedStoredCompany)
 
-        apiKeyHandler.revokeApiKeyForUserTypeAndRevokeBearerTokens(userType)
-
+        apiKeyHelper.revokeApiKeyForTechnicalUserAndResetAuthentication(technicalUser)
+        GlobalAuth.setBearerToken(apiKey.apiKey)
```
Comment (fsoffel) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/DataRetrievalViaApiKeyTest.kt:140]: I would prefer using a revoke only method here. What do you think?

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/auth/ApiKeyAuthenticationHelper.kt [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+    fun revokeApiKeyForTechnicalUserAndResetAuthentication(technicalUser: TechnicalUser): RevokeApiKeyResponse {
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(technicalUser)
+        val revokeApiKeyResponse = apiKeyManagerClient.revokeApiKey()
+        GlobalAuth.setBearerToken(null)
+        return revokeApiKeyResponse
+    }
+
+    fun resetAuthenticationAndValidateApiKey(receivedApiKey: String): ApiKeyMetaInfo {
+        GlobalAuth.setBearerToken(null)
```
Comment (fsoffel) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/auth/ApiKeyAuthenticationHelper.kt:34]: Is this only done to check the reachability of this endpoint without any authorities? Technically the bearer token could remain set. I just found it quite weird to read the name of this function so often in the DataRetrievalViaApiKeyTest.kt

File: dataland-backend/build.gradle.kts [hunk 7]
```diff
@@ -50,6 +50,7 @@ dependencies {
     kapt("org.springframework.boot:spring-boot-configuration-processor")
     implementation("org.springframework.boot:spring-boot-starter-security")
     testImplementation("org.springframework.boot:spring-boot-starter-test")
+    testImplementation("org.springframework.security:spring-security-test")
```
Comment (fsoffel) [inline dataland-backend/build.gradle.kts:53]: Why is that added? You did not touch the tests of this project.

File: dataland-keycloak-adapter/src/main/kotlin/org/dataland/keycloakAdapter/auth/DatalandRealmRole.kt (hunk 1 above)
Comment (MarcTM01) [reply dataland-keycloak-adapter/src/main/kotlin/org/dataland/keycloakAdapter/auth/DatalandRealmRole.kt:8]: No worries. From my point of view each enum constant only referrs to a single Role and not multiple Roles. Thats why I chose to use singulars.
